### PR TITLE
ci: Run linters with GithubActions

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Run lint on every push with Github Actions. Github Actions has now two linters:

1. Validating the Podspec: Fails when the Podspec is invalid
2. SwiftLint: Displays the swift lint warnings as check annotations

Furthermore, the code gets formatted and pushed if not properly formatted.

We can run this in parallel with Travis and it is a first step towards moving from Travis to Github Actions.

## :bulb: Motivation and Context

We want to know early when we violate linting rules.

## :green_heart: How did you test it?
With Github Actions.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] No breaking changes

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          style: file
+          inplace: True
+      - uses: EndBug/add-and-commit@v5
+        with:
+          author_name: Clang Robot
+          author_email: clang-robot@sentry.io
+          message: 'Committing clang-format changes'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}    
+
```
Comment (bruno-garcia) [inline .github/workflows/lint.yml:45]:
```suggestion
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      - uses: actions/checkout@v2
+      - name: Run clang-format
+        uses: DoozyX/clang-format-lint-action@v0.10
+        with:
+          source: '.'
+          extensions: 'h,cpp,m,c'
+          clangFormatVersion: 11
+          style: file
+          inplace: True
+      - uses: EndBug/add-and-commit@v5
```
Comment (bruno-garcia) [inline .github/workflows/lint.yml:38]: ❤️

**Review by bruno-garcia — COMMENTED** [review]
This is great stuff, I wish we get this lint/push on all repos.

swift lint doesn't push the fixes? build has some warnings!

**Comment by philipphofmann** [comment]
> swift lint doesn't push the fixes? build has some warnings!

No, it doesn't push the fixes because they are things as file too long or such, that you can't fix automatically, but I can add formatting the code as well for swift and committing if you messed it up.

**Review by bruno-garcia — APPROVED** [review]
Nice!

**Pull request merged**